rust/src/client/request.rs

```diff
@@ -1,7 +1,35 @@
+//! Provides functionality to enable clients to communicate with a XayNet
```

Comment:
the `request` module is private, therefore this documentation is invisible in the generated docs. i'd suggest to make it a subsection of the `client/mod.rs` docs.